Maintenance of oauth

fvdbeek opened the pull request:
The [emacs-oauth](https://github.com/psanford/emacs-oauth) repository has been archived (see also #8510), and I would like to adopt the oauth package.

riscy commented:
Thank you, I think this is very likely to go through but it would be good if we could involve @psanford; maybe the best case scenario is that the repository is officially transferred to @fvdbeek, so that redirects are intact?

For reference: https://github.com/melpa/melpa/wiki/Unmaintained-Packages-and-Forks

psanford commented:
I'm not interested in transferring this repository at this time. Please fork it if you would like to maintain it.

fvdbeek commented:
That's fine with me. As mentioned in #8510, I deleted the fork awaiting a transfer. Now the fork is in the air again, but because I am used to GitLab, I imported the repository and changed the recipe accordingly.

riscy commented:
Thank you for your prompt reply, @psanford.   @fvdbeek, although the original repo is archived, consider the second paragraph at the aforementioned [wiki page](https://github.com/melpa/melpa/wiki/Unmaintained-Packages-and-Forks) -- is there truly something that needs to be maintained?

fvdbeek commented:
First of all, I don't _need_ to take over maintenance. But yes, the package could use some improvement:
- [it has some unattended issues](https://github.com/psanford/emacs-oauth/issues), one fairly recent.
- my own [emacs-zotero](https://gitlab.com/fvdbeek/emacs-zotero) package depends on emacs-oauth, but it needs [advising a function](https://gitlab.com/fvdbeek/emacs-zotero/-/blob/811bd1f14b38c3dde3f80cd8a13490c9900de888/zotero-auth.el) to work.
- it bundles [an hmac-sha1 implemententation](https://github.com/psanford/emacs-oauth/blob/master/hmac-sha1.el) that conflicts with [another implementation with the exact same name in the flim package](https://github.com/wanderlust/flim/blob/2cf5a7891090faca8de725b1d3743dcedf233ea2/hmac-sha1.el).

riscy commented:
These are totally valid and well-annotated -- thank you.  I am inferring the original author (whose work I really do appreciate) has no intention of accepting/reviewing pull requests due to the package being archived, so we can go ahead in this case.

tarsius commented:
> it bundles [an hmac-sha1 implemententation](https://github.com/psanford/emacs-oauth/blob/master/hmac-sha1.el) that conflicts with [another implementation with the exact same name in the flim package](https://github.com/wanderlust/flim/blob/2cf5a7891090faca8de725b1d3743dcedf233ea2/hmac-sha1.el).

Do you still plan to fix that?

fvdbeek commented:
Sure, I am even doing a full rewrite of the library. But that requires rigorous testing before I will push it. This fix is simple enough however, so I just pushed this.

tarsius commented:
Thanks!
